posthog/models/subscription.py

```diff
@@ -123,28 +123,19 @@ def url(self):
 
     @property
     def summary(self):
-        human_frequency_map = {
-            "daily": "day",
-            "weekly": "week",
-            "monthly": "month",
-            "yearly": "year",
-        }
-
-        human_bysetpos_map = {
-            1: "first",
-            2: "second",
-            3: "third",
-            4: "fourth",
-            -1: "last",
-        }
-
-        frequency = human_frequency_map[self.frequency]
+        human_frequency = {"daily": "day", "weekly": "week", "monthly": "month", "yearly": "year"}.get(
+            self.frequency, ""
+        )
         if self.interval > 1:
-            frequency = f"{frequency}s"
-        summary = f"sent every {str(self.interval) + ' ' if self.interval > 1 else ''}{frequency}"
+            human_frequency = f"{human_frequency}s"
+
+        summary = f"sent every {str(self.interval) + ' ' if self.interval > 1 else ''}{human_frequency}"
 
         if self.byweekday and self.bysetpos:
-            summary += f" on the {human_bysetpos_map.get(self.bysetpos, '')} {self.byweekday[0].capitalize() if len(self.byweekday) == 1 else 'day'}"
+            human_bysetpos = {1: "first", 2: "second", 3: "third", 4: "fourth", -1: "last",}.get(self.bysetpos, "")
```

Comment:
To continue nitpicking :)

So now in the case of unknown values we'd say something like "sent every 1 
  on the    day"??

Reply:
Ha... GH unhappy with formatting multiple spaces

Reply:
Indeed we would :D Which is not super nice but also not technically possible with the current options available. I think the outcome of "weird summary" is probably better than crashing or not giving one at all. What do you think?

Reply:
That is beautiful and a small change we may be very grateful for in the future.  I like it.

Reply:
![aw-shucks](https://user-images.githubusercontent.com/984817/174740398-f313bc93-929c-48c6-be4b-5251e830cc60.gif)